spec/sections_spec.rb

```diff
@@ -22,6 +26,20 @@ class Page < SitePrism::Page
     it { is_expected.to respond_to(:has_single_section?) }
   end
 
+  describe '.section with class and block' do
+    before do
+      expect(subject).to receive(:find_first).and_return(:element)
+    end
+
+    it 'should be an instance of provided section class' do
+      expect(subject.section_with_a_block.class.ancestors).to include(SingleSection)
+    end
+
+    it 'should have elements from the block' do
```

Comment:
Should we therefore add an extra test `it 'has elements from the defined class'`

Then add an extra element definition after opening the class in this file on line8, call it whatever makes you happy. (Just add one)